src/Module/Settings/Display.php

```diff
@@ -224,10 +240,21 @@ protected function content(array $request = []): string
 			BBCode::PREVIEW_LARGE    => $this->t('Large Image'),
 		];
 
-		$network_timelines = $this->pConfig->get($uid, 'system', 'network_timelines', array_keys($this->getAvailableTimelines($uid, true)));
+		$bookmarked_timelines = $this->pConfig->get($uid, 'system', 'network_timelines', array_keys($this->getAvailableTimelines($uid, true)));
+		$enabled_timelines    = $this->pConfig->get($uid, 'system', 'enabled_timelines', array_keys($this->getAvailableTimelines($uid, false)));
+
 		$channel_languages = $this->pConfig->get($uid, 'channel', 'languages', [User::getLanguageCode($uid)]);
 		$languages         = $this->l10n->getAvailableLanguages(true);
-		$timelines         = $this->getAvailableTimelines($uid);
+
+		$timelines = [];
+		foreach ($this->getAvailableTimelines($uid) as $code => $timeline) {
+			$timelines[] = [
+				'label'        => $timeline->label,
+				'description'  => $timeline->description,
+				'enable'       => ["enable[$code]", '', in_array($code, $enabled_timelines)],
+				'bookmark'     => ["bookmark[$code]", '', in_array($code, $bookmarked_timelines)],
```

Comment:
Since the entities have the `code` property in common, you can drop the `$code` variable from the array key.
```suggestion
				'enable'       => ["enable[{$timeline->code}]", '', in_array($timeline->code, $enabled_timelines)],
				'bookmark'     => ["bookmark[{$timeline->code}]", '', in_array($timeline->code, $bookmarked_timelines)],
```

Reply:
Done